community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/DiskLayer.java

```diff
@@ -179,70 +183,164 @@ public int labelGetForName( String label )
     }
 
     @Override
-    public boolean nodeHasLabel( long nodeId, int labelId ) throws EntityNotFoundException
+    public boolean nodeHasLabel( StoreStatement statement, long nodeId, int labelId ) throws EntityNotFoundException
     {
-        try
-        {
-            return PrimitiveIntCollections.indexOf( nodeGetLabels( nodeId ), labelId ) != -1;
-        }
-        catch ( InvalidRecordException e )
+        try ( StoreSingleNodeCursor nodeCursor = statement.acquireSingleNodeCursor() )
         {
-            return false;
+            if ( nodeCursor.init( nodeId ).next() )
```

Comment:
can't we initialize the cursor when acquiring it? something like `statement.acquireSingleNodeCursor( nodeId )`